sre-recipes/recipe.py

```diff
@@ -40,3 +42,76 @@ def verify(self):
         Verifies that the user of the recipe found the root cause
         of the breakage
         """
+
+    @abc.abstractmethod
+    def hint(self):
+        """
+        Provides a hint about the root cause of the issue
+        """
+
+    @staticmethod
+    def _run_command(command):
+        """Runs the given command and returns any output and error"""
+        process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return output, error
+
+    @staticmethod
+    def _auth_cluster(cluster="APP"):
+        """ Authenticates for kubectl commands. """
+        logging.info('Authenticating cluster')
+        get_project_command = 'gcloud config list --format value(core.project)'
+        project_id, error = Recipe._run_command(get_project_command)
+        project_id = project_id.decode("utf-8").replace('"', '')
+        if project_id == '':
+            print('No project found.')
+            logging.error('Could not authenticate cluster. No project ID found.')
+            exit(1)
+        name = "cloud-ops-sandbox"
+        if cluster == 'LOADGEN':
+            name = 'loadgenerator'
+        zone_command = 'gcloud container clusters list --filter name:{} --project {} --format value(zone)'.format(name, project_id)
+        zone, error = Recipe._run_command(zone_command)
+        zone = zone.decode("utf-8").replace('"', '')
+        if zone == '':
+            print('Failed to set up recipe. No cluster for {} was found.'.format(name))
```

Comment:
nit: it would probably be better to use try blocks and exceptions so you can handle these errors in one place instead of disrupting the flow of the code. But with only 2 I don't think it matters too much